Microsoft.Toolkit/Collections/ObservableGroup.cs

```diff
@@ -16,8 +17,10 @@ namespace Microsoft.Toolkit.Collections
     /// <typeparam name="TKey">The type of the group key.</typeparam>
     /// <typeparam name="TValue">The type of the items in the collection.</typeparam>
     [DebuggerDisplay("Key = {Key}, Count = {Count}")]
-    public sealed class ObservableGroup<TKey, TValue> : ObservableCollection<TValue>, IGrouping<TKey, TValue>, IReadOnlyObservableGroup
+    public class ObservableGroup<TKey, TValue> : ObservableCollection<TValue>, IGrouping<TKey, TValue>, IReadOnlyObservableGroup
     {
+        private TKey key;
```

Comment:
Nit: could you move this field declaration above the `Key` property below (leaving one blank line above it), to be consistent with other explicit field-backed properties in the MVVM Toolkit? See here:

https://github.com/windows-toolkit/WindowsCommunityToolkit/blob/6583c81505cef338130a8b79e9f29878789b62b5/Microsoft.Toolkit.Mvvm/Input/AsyncRelayCommand.cs#L86-L89